warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -442,6 +431,19 @@ private DateIndexHelper getDateIndexHelper(final ShardQueryConfiguration config)
         }
         
         stopwatch.stop();
+        this.plannedScript = newQueryString;
+        
+        Tuple2<CloseableIterable<QueryPlan>,Boolean> queryRanges = getQueryRanges(scannerFactory, metadataHelper, config, queryTree);
+        
+        // a full table scan is required if
+        final boolean isFullTable = queryRanges.second();
+        
+        // abort if we cannot handle full table scans
+        if (isFullTable && !config.getFullTableScanEnabled()) {
+            PreConditionFailedQueryException qe = new PreConditionFailedQueryException(DatawaveErrorCode.FULL_TABLE_SCAN_REQUIRED_BUT_DISABLED);
+            throw new FullTableScansDisallowedException(qe);
```

Comment:
Won't the query plan fail to be set if this exeception is thrown?

Reply:
Oh, you're right, it would just be whatever "config.setQueryString(jexlQueryString);" is from the ShardQueryLogic.

I can update it to set the plan there before that exception is thrown.